cohortextractor/query_language.py

```diff
@@ -170,6 +170,27 @@ def sum(self, column):  # noqa: A003
     def aggregate(self, function, column):
         return ValueFromAggregate(self, function, column)
 
+    def imd_rounded_as_of(self, reference_date):
```

Comment:
This adds a table-specifc method in the same way that Dave's age_as_of does.  I don't really like it, but it's the simplest thing for now, pending decisions on adding named tables etc.

Reply:
I agree about not liking it and about it being the best thing to do for now!